src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbTest.java

```diff
@@ -18,23 +22,33 @@
 
     @ParameterizedTest(name = "{arguments}")
     @ArgumentsSource(DefaultArgumentProvider.class)
-    public void testListTables_noMtContext(TestArgument testArgument) {
+    public void testListTables_noContext(TestArgument testArgument) {
         MT_CONTEXT.setContext(null);
         List<String> allTables = testArgument.getAmazonDynamoDb().listTables().getTableNames();
         assertTrue(allTables.size() > 0);
+    }
 
-        // test exclusive start key
-        List<String> subset = testArgument.getAmazonDynamoDb().listTables(allTables.get(1)).getTableNames();
-        assertEquals(allTables.size(), subset.size() + 2);
-
-        // test pagination
-        for (int i = 0; i < allTables.size() - 2; i++) {
-            List<String> tables = testArgument.getAmazonDynamoDb().listTables(allTables.get(i), 2)
-                .getTableNames();
-            assertEquals(2, tables.size());
-            assertEquals(tables.get(0), allTables.get(i + 1));
-            assertEquals(tables.get(1), allTables.get(i + 2));
-        }
-
+    @ParameterizedTest(name = "{arguments}")
+    @ArgumentsSource(DefaultArgumentProvider.class)
+    public void testListTables_noContext_pagination(TestArgument testArgument) {
+        MT_CONTEXT.setContext(null);
+        List<String> allTables = Lists.newArrayList();
+        Set<String> allTableSet = new HashSet<>();
+        ListTablesResult listTablesResult;
+        String exclusiveStartKey = null;
+        int lastBatchSize;
+        int limit = 5;
+        do {
+            listTablesResult = testArgument.getAmazonDynamoDb().listTables(exclusiveStartKey, limit);
+            lastBatchSize = listTablesResult.getTableNames().size();
+            assertTrue(listTablesResult.getTableNames().size() > 0 || lastBatchSize == limit,
+                "Should not have gotten empty table set, or last result set should have been full");
```

Comment:
this assertion should address the bug BenB found of always populating last evaluated table name, which causes a client to need to pull an extra empty batch at the end. now that only happens if the previous batch is full, which matches dynamos behavior